pytorch_lightning/trainer/trainer.py

```diff
@@ -679,23 +679,9 @@ def training_tqdm_dict(self):
         """Read-only for tqdm metrics.
         :return:
         """
-        tqdm_dict = {
-            'loss': '{0:.3f}'.format(self.avg_loss),
-            'batch_idx': '{}'.format(self.batch_idx),
-        }
+        ref_model = self.model if not self.data_parallel else self.model.module
 
-        if self.truncated_bptt_steps is not None:
-            tqdm_dict['split_idx'] = self.split_idx
-
-        if self.logger is not None and self.logger.version is not None:
-            tqdm_dict['v_num'] = self.logger.version
-
-        tqdm_dict.update(self.tqdm_metrics)
-
-        if self.on_gpu:
-            tqdm_dict['gpu'] = '{}'.format(torch.cuda.current_device())
-
-        return tqdm_dict
+        return dict(**ref_model.get_tqdm_dict(), **self.tqdm_metrics)
```

Comment:
I wold go with
```python
if self.tqdm_metrics == None:
    return ref_model.get_tqdm_dict()
else:
    return self.tqdm_metrics
```
so that that the default dict can be completely overloaded

Reply:
But it is already can be completely overloaded :)
`self.tqdm_metrics` is not the default, it is totally determined by `progress_bar` dict from the `training_end()` and `validation_end()`. And `model.get_tqdm_dict()` adds some additional information like version number, gpu, etc. With this PR it is also totally controlled by the user.

So if you return an empty dict in `model.get_tqdm_dict()` and have no `progress_bar` dicts you will have a tqdm progress bar without any additional info.

Reply:
right, sorry for the noise, I thought that `get_tqdm_dict` was not exposed. Is it discoverable? Should something be added somewhere in the docs?

Reply:
Well, now it can be found as one of the methods of `LightningModule`. Maybe we can add some information about how progress bar is composed, but I don't understand where it can be inserted in the current docs.